backend/src/routes/user.ts

```diff
@@ -232,7 +263,15 @@ router.get("/mentor/:userId", [verifyAuthToken], async (req: Request, res: Respo
         graduationYear,
         calendlyLink,
         topicsOfExpertise,
+        pairingIds,
       } = mentor;
+      let whyPaired = "N/A";
+      for(const i in pairingIds) {
```

Comment:
If we can't use the method of lookup with aggregate here, then lets atleast change this to a parallelized query call. 

Running a query inside a for loop will run each query one after the other. Since these queries are independent of each other, we can leverage that. Look into `Promise.all()` to parallelize this -
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
https://sharegpt.com/c/QpUDZo9